hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -118,6 +118,15 @@ object HoodieSparkSqlWriter {
 
       operation = WriteOperationType.INSERT
     }
+    // If no record key field and precombine field is provided, assume it's an append-only workload and do bulk insert
+    if (!hoodieConfig.contains(RECORDKEY_FIELD) && !hoodieConfig.contains(PRECOMBINE_FIELD) &&
+      operation == WriteOperationType.UPSERT) {
+
+      log.warn(s"$RECORDKEY_FIELD and $PRECOMBINE_FIELD is not specified " +
+        s"overriding the $OPERATION to be $BULK_INSERT_OPERATION_OPT_VAL")
+
+      operation = WriteOperationType.BULK_INSERT
```

Comment:
but we still populate meta fields? So for users doing this sort of benchmarking out of box, Hudi will appear slower?

Reply:
you're right. i'm going to disable meta fields by default for this mode